File: generate_data.py
import json
import random
from datetime import datetime, timedelta

products = ['Fiber Internet 300 Mbps', '5G Unlimited Mobile Plan', 'Fiber Internet 1 Gbps', 'Business Internet 500 Mbps', 'VoIP Corporate Package']
statuses = ['Pending', 'In progress', 'Completed']
countries = ['United States', 'Canada', 'Australia', 'Singapore', 'Hong Kong', 'United Kingdom', 'Germany', 'France']
representatives = ['Mr. Michael Harris', 'Mr. Ryan Cooper', 'Ms. Olivia Carter', 'Mr. Lucas Martin']
names = ['John Doe', 'Jane Smith', 'Alice Johnson', 'Robert Brown', 'Emily Davis', 'Michael Wilson', 'Sarah Moore', 'David Taylor', 'Laura Anderson', 'James Thomas', 'William Jackson', 'Linda White', 'Barbara Harris', 'Elizabeth Martin', 'Richard Thompson']

orders = []
now = datetime.now()

for i in range(1, 61):
    qty = random.randint(1, 5)
    price = random.randint(50, 250)
    random_days_ago = random.randint(0, 90)
    order_date = now - timedelta(days=random_days_ago)
    
    full_name = random.choice(names)
    first_name = full_name.split(' ')[0]
    last_name = full_name.split(' ')[1] if ' ' in full_name else 'Smith'
    
    order = {
        "id": f"ORD-{i:06d}",
        "firstName": first_name,
        "lastName": last_name,
        "customerName": f"{first_name} {last_name}",
        "email": f"{first_name.lower()}.{last_name.lower()}@example.com",
        "phone": f"555-01{random.randint(0, 99):02d}",
        "address": f"{random.randint(1, 9999)} Main St",
        "city": "Metropolis",
        "state": "NY",
        "zip": "10001",
        "country": random.choice(countries),
        "product": random.choice(products),
        "quantity": qty,
        "unitPrice": price,
        "totalAmount": qty * price,
        "status": random.choice(statuses),
        "createdBy": f"{random.choice(representatives)} (for {first_name} {last_name})",
        "orderDate": order_date.isoformat() + "Z"
    }
    orders.append(order)

print(json.dumps(orders, indent=2))

File: generate_data.js
const products = ['Fiber Internet 300 Mbps', '5G Unlimited Mobile Plan', 'Fiber Internet 1 Gbps', 'Business Internet 500 Mbps', 'VoIP Corporate Package'];
const statuses = ['Pending', 'In progress', 'Completed'];
const countries = ['United States', 'Canada', 'Australia', 'Singapore', 'Hong Kong', 'United Kingdom', 'Germany', 'France'];
const representatives = ['Mr. Michael Harris', 'Mr. Ryan Cooper', 'Ms. Olivia Carter', 'Mr. Lucas Martin'];
const names = ['John Doe', 'Jane Smith', 'Alice Johnson', 'Robert Brown', 'Emily Davis', 'Michael Wilson', 'Sarah Moore', 'David Taylor', 'Laura Anderson', 'James Thomas', 'William Jackson', 'Linda White', 'Barbara Harris', 'Elizabeth Martin', 'Richard Thompson'];

const orders = [];
const now = new Date();

for (let i = 1; i <= 60; i++) {
    const qty = Math.floor(Math.random() * 5) + 1;
    const price = Math.floor(Math.random() * 200) + 50;
    const randomDaysAgo = Math.floor(Math.random() * 90);
    const orderDate = new Date(now.getTime() - randomDaysAgo * 24 * 60 * 60 * 1000);
    
    const firstName = names[Math.floor(Math.random() * names.length)].split(' ')[0];
    const lastName = names[Math.floor(Math.random() * names.length)].split(' ')[1] || 'Smith';
    
    orders.push({
        id: 'ORD-' + i.toString().padStart(6, '0'),
        firstName: firstName,
        lastName: lastName,
        customerName: `${firstName} ${lastName}`,
        email: `${firstName.toLowerCase()}.${lastName.toLowerCase()}@example.com`,
        phone: '555-01' + Math.floor(Math.random() * 100).toString().padStart(2, '0'),
        address: `${Math.floor(Math.random() * 9999)} Main St`,
        city: 'Metropolis',
        state: 'NY',
        zip: '10001',
        country: countries[Math.floor(Math.random() * countries.length)],
        product: products[Math.floor(Math.random() * products.length)],
        quantity: qty,
        unitPrice: price,
        totalAmount: qty * price,
        status: statuses[Math.floor(Math.random() * statuses.length)],
        createdBy: `${representatives[Math.floor(Math.random() * representatives.length)]} (for ${firstName} ${lastName})`,
        orderDate: orderDate.toISOString(),
        ownerId: 'system'
    });
}

console.log(JSON.stringify(orders, null, 2));
